src/anomalib/data/image/mvtec_loco.py

```diff
@@ -0,0 +1,597 @@
+
+
+def load_saturation_config(config_path: str | Path) -> dict[int, Any] | None:
+    """Load saturation configurations from a JSON file.
+
+    Args:
+        config_path (str | Path): Path to the saturation configuration file.
+
+    Returns:
+        Dict | None: A dictionary with pixel values as keys and the corresponding configurations as values.
+            Return None if the config file is not found.
+
+    Example JSON format in the file:
+    [
+        {
+            "defect_name": "1_additional_pushpin",
+            "pixel_value": 255,
+            "saturation_threshold": 6300,
+            "relative_saturation": false
+        },
+        {
+            "defect_name": "2_additional_pushpins",
+            "pixel_value": 254,
+            "saturation_threshold": 12600,
+            "relative_saturation": false
+        },
+        ...
+    ]
+    """
+    try:
+        config_path = validate_path(config_path)
+        with Path.open(config_path) as file:
+            configs = json.load(file)
+        # Create a dictionary with pixel values as keys
+        return {conf["pixel_value"]: conf for conf in configs}
+    except FileNotFoundError:
+        logger.warning("The saturation config file %s does not exist. Returning None.", config_path)
+        return None
+
+
+def _merge_gt_mask(
+    root: str | Path,
+    extensions: Sequence[str] = IMG_EXTENSIONS,
+    gt_merged_dir: str | Path = GT_MERGED_DIR,
+) -> None:
+    """Merges ground truth masks within specified directories and saves the merged masks.
+
+    Args:
+        root (str | Path): Root directory containing the 'ground_truth' folder.
+        extensions (Sequence[str]): Allowed file extensions for ground truth masks.
+                                    Default is IMG_EXTENSIONS.
+        gt_merged_dir (str | Path]): Directory where merged masks will be saved.
+                                     Default is GT_MERGED_DIR.
+
+    Returns:
+        None
+
+    Example:
+        >>> _merge_gt_mask('path/to/breakfast_box/')
+
+        This function reads ground truth masks from the specified directories, merges them into
+        a single mask for each corresponding images (e.g. merge 059/000.png and 059/001.png into 059.png),
+        and saves the merged masks in the default GT_MERGED_DIR directory.
+
+        Note: The merged masks are saved with the same filename structure as the corresponding anomalous image files.
+    """
+    root = Path(root)
+    gt_mask_paths = {f.parent for f in root.glob("ground_truth/**/*") if f.suffix in extensions}
+
+    for mask_path in gt_mask_paths:
+        # Merge each mask inside mask_path into a single mask
+        merged_mask = None
+        for mask_file in mask_path.glob("*"):
+            if mask_file.suffix in extensions:
+                mask = cv2.imread(str(mask_file), cv2.IMREAD_UNCHANGED)
+                if merged_mask is None:
+                    merged_mask = np.zeros_like(mask)
```

Comment:
Do we assume that masks are here not overlapping. Is this assumption correct for all cases?

Reply:
Oh, that's a great observation! After checking all categories, it seems there's a possibility of overlapping only in the 'screw_bag' category. Although the overlapping annotations might not be easily visible, there are indeed small overlaps at the borders between the regions. I believe this implies that we can't simply merge the masks into a single mask, right? Even if such overlaps are unintentional, we lack information about which regions the overlapping annotations are supposed to belong to. It seems we need to pass a list of tensors instead of a single merged mask to handle the possibility of such cases.

For example:
![000](https://github.com/openvinotoolkit/anomalib/assets/5786636/a4dec9d3-fb56-4a61-8ded-cdca693cb4fe)
![001](https://github.com/openvinotoolkit/anomalib/assets/5786636/109804d9-f0d9-4d90-bead-b89406c285d2)

Reply:
> It seems we need to pass a list of tensors instead of a single merged mask to handle the possibility of such cases.
I believe this would indeed be the best option for all these problems, or at least something similar to this approach. It'd also avoid problem if somewhere in the future a new such dataset appears that would break our assumptions.

Reply:
> > It seems we need to pass a list of tensors instead of a single merged mask to handle the possibility of such cases.
> 
> ...It'd also avoid problem if somewhere in the future a new such dataset appears that would break our assumptions.

This example looks like an imperfection in the annotations. 

But, yeah, overlap of logical masks sounds like something that will eventually happen. So I agree we should assume overlap as possible. 

Does it require changing the interface (a method would now require accepting tensor + list of tensors) in many places, @willyfh ?

---

just for the sake of documenting that example in https://github.com/openvinotoolkit/anomalib/pull/1686#discussion_r1480649912 

MVTec LOCO, Sec 3.1 and Tab 12
A screw bag contains exactly two washers, two nuts, one long screw, and one short screw.

MVTecLOCO/screw_bag/ground_truth/logical_anomalies/129/
000.png --> white = 248 --> 2_additional_nuts
001.png --> white = 240 --> 2_missing_washers

Reply:
I have finally managed to pass the list of tensors (after a painfully trial and error process 😅). We need to update collate function to handle the unequal length of list in the batch, and also update the `_outputs_to_device`.

https://github.com/openvinotoolkit/anomalib/blob/2cdeb82eb79348a23325fb3d873862775366faa2/src/anomalib/data/base/datamodule.py#L45-L48

https://github.com/openvinotoolkit/anomalib/blob/2cdeb82eb79348a23325fb3d873862775366faa2/src/anomalib/callbacks/metrics.py#L200-L202